core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,611 @@
+      const OtherMapping &rhs) {
+    using src_t          = typename OtherMapping::layout_type;
+    using dst_t          = layout_type;
+    constexpr size_t rnk = mdspan_type::rank();
+    if constexpr (!std::is_same_v<src_t, dst_t>) {
+      if constexpr (Impl::is_layout_left_padded<dst_t>::value) {
+        if constexpr (std::is_same_v<src_t, layout_stride>) {
+          index_type stride = 1;
+          for (size_t r = 0; r < rnk; r++) {
+            if (rhs.stride(r) != stride)
+              Kokkos::abort("View assignment must have compatible layouts");
+            if constexpr (rnk > 1)
+              stride *= (r == 0 ? rhs.stride(1) : rhs.extents().extent(r));
+          }
+        }
+      }
+      if constexpr (Impl::is_layout_right_padded<dst_t>::value) {
+        if constexpr (std::is_same_v<src_t, layout_stride>) {
+          index_type stride = 1;
+          if constexpr (rnk > 0) {
+            for (size_t r = rnk; r > 0; r--) {
+              if (rhs.stride(r - 1) != stride)
+                Kokkos::abort("View assignment must have compatible layouts");
+              if constexpr (rnk > 1)
+                stride *= (r == rnk ? rhs.stride(r - 2)
+                                    : rhs.extents().extent(r - 1));
+            }
+          }
+        }
+      }
+      if constexpr (std::is_same_v<dst_t, layout_left>) {
+        if constexpr (std::is_same_v<src_t, layout_stride>) {
+          index_type stride = 1;
+          for (size_t r = 0; r < rnk; r++) {
+            if (rhs.stride(r) != stride)
+              Kokkos::abort("View assignment must have compatible layouts");
+            stride *= rhs.extents().extent(r);
+          }
+        } else if constexpr (Impl::is_layout_left_padded<src_t>::value &&
+                             rnk > 1) {
+          if (rhs.stride(1) != rhs.extents().extent(0))
+            Kokkos::abort("View assignment must have compatible layouts");
+        }
+      }
+      if constexpr (std::is_same_v<dst_t, layout_right>) {
+        if constexpr (std::is_same_v<src_t, layout_stride>) {
+          index_type stride = 1;
+          if constexpr (rnk > 0) {
+            for (size_t r = rnk; r > 0; r--) {
+              if (rhs.stride(r - 1) != stride)
+                Kokkos::abort("View assignment must have compatible layouts");
+              stride *= rhs.extents().extent(r - 1);
+            }
+          }
+        } else if constexpr (Impl::is_layout_right_padded<src_t>::value &&
+                             rnk > 1) {
+          if (rhs.stride(rnk - 2) != rhs.extents().extent(rnk - 1))
+            Kokkos::abort("View assignment must have compatible layouts");
+        }
+      }
+    }
+  }
+
+ public:
+  KOKKOS_DEFAULTED_FUNCTION constexpr BasicView() = default;
+
+  // Constructors from mdspan
+  KOKKOS_DEFAULTED_FUNCTION constexpr BasicView(const BasicView &) = default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr BasicView(BasicView &&)      = default;
```

Comment:
```suggestion
```

Reply:
Are these getting generated if we have ` BasicView(const BasicView<OtherT, OtherE, OtherL, OtherA> &other,`? I am not 100% sure.

Reply:
Yes, I think so https://godbolt.org/z/6E3KdPTnq